doc/user/content/integrations/terraform.md

```diff
@@ -0,0 +1,117 @@
+---
+title: "Terraform Materialize Provider"
+description: "Create and manage Materialize objects with the Terraform Materialize provider"
+menu:
+  main:
+    parent: "integrations"
+    weight: 11
+    name: "Terraform Materialize Provider"
+---
+
+Terraform is an infrastructure-as-code tool that allows you to manage your
+resources in a declarative configuration language. Materialize maintains a
+Terraform provider to help you configure Materialize resources like
+connections, sources, and database objects.
+
+## Use cases
+
+Materialize objects can be created through standard SQL queries, but managing
+more than a few components can be difficult to maintain. The Terraform
+Materialize provider allows users who are not well-versed in SQL to maintain
+objects in an organization. A few specific use cases are captured in the
+sections below.
+
+### MSK or self-managed Kafka and PrivateLink
+
+To get data into Materialize, you need a connection to allow your data source to
+communicate with Materialize. One option to connect securely to Materialize is
+AWS PrivateLink.
+
+The [AWS MSK PrivateLink](https://github.com/MaterializeInc/terraform-aws-msk-privatelink) and [AWS Kafka PrivateLink](https://github.com/MaterializeInc/terraform-aws-kafka-privatelink) modules allow you to manage
+your connection to Materialize in a single configuration. With these modules,
+you can create a secure connection and add Materialize [resource blocks](https://registry.terraform.io/providers/MaterializeInc/materialize/latest/docs/resources/connection_aws_privatelink) to allow
+Materialize to communicate with the PrivateLink endpoint.
+
+### EC2 SSH bastion host
+
+Another method for source connection is to use a bastion host to allow SSH
+communication to and from Materialize.
+
+The [EC2 SSH bastion module](https://github.com/MaterializeInc/terraform-aws-ec2-ssh-bastion) allows you to configure an EC2 instance with security groups and an SSH keypair. After using the module, you can configure the `materialize_connection_ssh_tunnel` resource, allowing Materialize an end-to-end connection to your source.
+
+### Postgres RDS
+
+You can also create an RDS instance from which you can track and propagate
+changes. The [AWS RDS Postgres module](https://github.com/MaterializeInc/terraform-aws-rds-postgres) creates a VPC, security groups, and an RDS instance in AWS. After you run the module, you can create a secret, connection and source with the Materialize Terraform provider for an end-to-end connection to this instance as a new source.
```

Comment:
I like this section, but it feels out of place. Does it make sense to discuss these separately from the mz terraform provider?

Reply:
I know the modules are not immediately related to the provider itself, but I think it adds value to see prebuilt examples. Do you think it would make more sense to have an actual end-to-end config with the Materialize provider resources included so it's a complete example?

Reply:
Added some info to differentiate modules from the provider and hopefully made that clearer. Let me know if this makes more sense now @sjwiesman